pkg/obfuscator/tracker.go

```diff
@@ -54,18 +54,23 @@ func (s *SimpleTracker) AddReplacement(original string, replacement string) {
 	s.mapping[original] = replacement
 }
 
-func (s *SimpleTracker) GenerateIfAbsent(key string, generator GenerateReplacement) string {
+func (s *SimpleTracker) GenerateIfAbsent(key string, generator GenerateReplacement) (string, bool) {
 	s.lock.Lock()
 	defer s.lock.Unlock()
+	// returning the empty string if the replacement is already present
+	// GenerateIfAbsent only should generate the replacement if the key is a new one
+	// This helps in avoiding the addition of case-sensitive alternatives to the report
```

Comment:
why is that a problem? I don't it it makes sense to add such a return value here. Can you explain why this is necessary instead of making AddReplacement idempotent?

Reply:
plus, I think that way you are also circumventing an important check that we don't add the same key with a different value multiple times.

Reply:
Lets say the same MAC address has been observed twice, one containing the upper case and another with lower case alphabets. Ex: "2C:54:91:88:C9:E3 can also be displayed as 2c:54:91:88:c9:e3"
When we call the Generate function, we normalize the key string to the uppercase and call the function which is "2C:54:91:88:C9:E3"
Though the substituted value remains the same("xxx-mac-000001-xxx"), both key-value replacement pairs would be appended to the report mapping which I feel is ambiguous. 
{
"2C:54:91:88:C9:E3": "xxx-mac-000001-xxx",
"2c:54:91:88:c9:e3": "xxx-mac-000001-xxx"
}
Hence, I have given a thorough thought and felt this extra return (a boolean) helps in verifying if the data is already added to the report and if not, can be added.